avro-builder/builder-spi/src/main/java/com/linkedin/avroutil1/builder/plugins/BuilderPluginContext.java

```diff
@@ -45,19 +45,27 @@ public void run() throws Exception {
       throw new IllegalStateException("run() has already been invoked");
     }
 
-    //"seal" any internal state to prevent plugins from trying to do weird things during execution
+    // "seal" any internal state to prevent plugins from trying to do weird things during execution
     sealed = true;
 
-    int operationCount = operations.stream().collect(StreamUtil.toParallelStream(op -> {
-      try {
-        op.run(operationContext);
-      } catch (Exception e) {
-        throw new IllegalStateException("Exception running operation", e);
-      }
+    if (!operations.isEmpty()) {
+      long operationStart = System.currentTimeMillis();
+      final int parallelism = Math.min(operations.size(), 5);
```

Comment:
where is this magic number 5 coming from? Also, what is the rationale for previous hard-coded 2?

Reply:
These plugin operations are generally heavily I/O bound so it is ok to kick off upto 5 threads in parallel. Earlier I hardcoded to 2 because I thought most use cases will have at most 2 plugins, but I realized that avro-schemas itself has 3, so rather than bound it by plugin count, I decided to bound it by hardware overhead.